internal/testing/replay/replay.go

```diff
@@ -16,168 +16,13 @@
-	cur := 0
-	r.SetMatcher(func(r *http.Request, i cassette.Request) bool {
-		if r.Method != i.Method {
-			t.Fatalf("mismatched Method at request #%d; got %q want %q", cur, r.Method, i.Method)
-		}
-		gotURL := r.URL.String()
-		wantURL := i.URL
-		for _, re := range matcher.URLScrubbers {
-			gotURL = re.ReplaceAllString(gotURL, "")
-			wantURL = re.ReplaceAllString(wantURL, "")
-		}
-		if gotURL != wantURL {
-			t.Fatalf("mismatched URL at request #%d; got\n%q\nwant\n%q", cur, gotURL, wantURL)
-		}
-		for _, header := range matcher.Headers {
-			got := r.Header.Get(header)
-			want := i.Headers.Get(header)
-			if got != want {
-				t.Fatalf("mismatched HTTP header %q header at request #%d; got %q want %q", header, cur, got, want)
-			}
-		}
-		var b bytes.Buffer
-		if r.Body != nil {
-			if _, err := b.ReadFrom(r.Body); err != nil {
-				t.Fatalf("couldn't read request body: %v", err)
-			}
-		}
-		r.Body = ioutil.NopCloser(&b)
-		gotBody := b.String()
-		wantBody := i.Body
-		for _, re := range matcher.BodyScrubbers {
-			gotBody = re.ReplaceAllString(gotBody, "")
-			wantBody = re.ReplaceAllString(wantBody, "")
-		}
-		if gotBody != wantBody {
-			t.Fatalf("mismatched HTTP body at request #%d", cur)
-		}
-
-		// We've got a match!
-		t.Logf("matched request #%d (%s %s)", cur, i.Method, i.URL)
-		cur++
-		return true
-	})
-	return r, func() {
-		if err := r.Stop(); err != nil {
-			fmt.Println(err)
-		}
-		if mode == recorder.ModeRecording {
-			if err := scrubRecording(path, matcher.DropRequestHeaders, matcher.DropResponseHeaders); err != nil {
-				t.Errorf("failed to scrub recording: %v", err)
-			}
-		}
-	}, nil
-}
-
-// scrubRecording scrubs the Authorization header.
-func scrubRecording(filepath string, dropRequestHeaders, dropResponseHeaders *regexp.Regexp) error {
-	c, err := cassette.Load(filepath)
-	if err != nil {
-		if os.IsNotExist(err) {
-			// Nothing to scrub!
-			return nil
-		}
-		return err
-	}
-
-	keep := make([]*cassette.Interaction, 0, len(c.Interactions))
-	c.Mu.Lock()
-	for _, action := range c.Interactions {
-		if action.Response.Code == http.StatusTooManyRequests {
-			// Too Many Requests, client will retry; it's safe
-			// to drop these for the replay.
-			continue
-		}
-		// Always drop the Authorization request header and the Duration
-		// response header.
-		action.Request.Headers.Del("Authorization")
-		action.Response.Headers.Del("Duration")
-
-		// Drop custom headers.
-		if dropRequestHeaders != nil {
-			for header := range action.Request.Headers {
-				if dropRequestHeaders.MatchString(header) {
-					action.Request.Headers.Del(header)
-				}
-			}
-		}
-		if dropResponseHeaders != nil {
-			for header := range action.Response.Headers {
-				if dropResponseHeaders.MatchString(header) {
-					action.Response.Headers.Del(header)
-				}
-			}
-		}
-		keep = append(keep, action)
-	}
-	c.Interactions = keep
-	c.Mu.Unlock()
-	return c.Save()
-}
-
```

Comment:
Can we move `NewGCPDialOptions` to `setup.go` and just delete this `replay` package entirely?

Reply:
(feel free to do in a separate PR)

Reply:
Yep, have separate PR in progress for that